Sources/Commands/SwiftPackageRegistryTool.swift

```diff
@@ -150,6 +140,221 @@ public struct SwiftPackageRegistryTool: ParsableCommand {
+            let authorizationWriter = authorizationProvider as? AuthorizationWriter
+            if saveChanges, authorizationWriter == nil {
+                throw StringError("Credential storage must be writable")
+            }
+
+            // Save in cache so we can try the credentials and persist to storage only if login succeeds
+            try tsc_await { callback in
+                authorizationWriter?.addOrUpdate(
+                    for: url,
+                    user: storeUsername,
+                    password: storePassword,
+                    persist: false,
+                    callback: callback
+                )
+            }
+            
+            // `url` can either be base URL of the registry, in which case the login API
+            // is assumed to be at /login, or the full URL of the login API.
+            var loginAPIPath: String?
+            if !url.path.isEmpty, url.path != "/" {
+                loginAPIPath = url.path
+            }
+            
+            guard let loginURL = URL(string: "https://\(host)\(loginAPIPath ?? "/login")") else {
+                throw RegistryConfigurationError.invalidURL(self.url)
+            }
+            
+            let configuration = try getRegistriesConfig(swiftTool)
+            
+            // Build a RegistryConfiguration with the given authentication settings
+            var registryConfiguration = configuration.configuration
+            registryConfiguration.registryAuthentication[host] = .init(type: authenticationType, loginAPIPath: loginAPIPath)
+
+            // Build a RegistryClient to test login credentials (fingerprints don't matter in this case)
+            let registryClient = RegistryClient(
+                configuration: registryConfiguration,
+                fingerprintStorage: .none,
+                fingerprintCheckingMode: .strict,
+                authorizationProvider: authorizationProvider
+            )
+
+            // Try logging in
+            try tsc_await { callback in
+                registryClient.login(
+                    url: loginURL,
+                    timeout: .seconds(3),
+                    observabilityScope: swiftTool.observabilityScope,
+                    callbackQueue: .sharedConcurrent,
+                    completion: callback
+                )
+            }
+            print("Login successful.")
+            
+            // Login successful. Persist credentials to storage.
+            
+            let osStore = !(authorizationWriter is NetrcAuthorizationProvider)
+            
+            // Prompt if writing to .netrc and --no-confirm is not set
+            if !osStore, !self.noConfirm {
+                print("""
+
+                WARNING: Secure credential storage is not supported on this platform.
+                Your credentials will be written out to .netrc.
+                """)
+                print("Continue? (Y/N): ")
+                guard readLine()?.lowercased() == "y" else {
```

Comment:
@neonichu @abertelrud is this the recommended way to read input?

Reply:
For plugin we do this:

```swift
                            // We can ask the user directly, so we do so.
                            let query = "Allow this plugin to write to the package directory?"
                            swiftTool.outputStream.write("\(problem)\n\(reason)\n\(query) (yes/no) ".utf8)
                            swiftTool.outputStream.flush()
                            let answer = readLine(strippingNewline: true)
                            // Throw an error if we didn't get permission.
                            if answer?.lowercased() != "yes" {
                                throw ValidationError("Plugin was denied permission to write to the package directory.")
                            }
```

Updating valid answers to yes/no and adding `strippingNewline: true` to `readLine` call.